lib/plugins/aws/package/compile/events/stream.js

```diff
@@ -273,6 +279,14 @@ class AwsCompileStreamEvents {
               };
             }
 
+            if (event.stream.patterns != null) {
```

Comment:
Any reason to not do `if(event.stream.patterns)` here?

Reply:
Just following previous checks pattern done in this file :
```ts
if (event.stream.batchWindow != null) {
  streamResource.Properties.MaximumBatchingWindowInSeconds = event.stream.batchWindow;
}

if (event.stream.maximumRetryAttempts != null) {
  streamResource.Properties.MaximumRetryAttempts = event.stream.maximumRetryAttempts;
}

if (event.stream.bisectBatchOnFunctionError != null) {
  streamResource.Properties.BisectBatchOnFunctionError =
    event.stream.bisectBatchOnFunctionError;
}

if (event.stream.functionResponseType != null) {
  streamResource.Properties.FunctionResponseTypes = [event.stream.functionResponseType];
}
```

I'm fine with changing to `if(event.stream.patterns)` :)